components/script/dom/webglframebuffer.rs

```diff
@@ -270,21 +270,41 @@ impl WebGLFramebuffer {
         let has_s = s.is_some();
         let has_zs = zs.is_some();
         let attachments = [&*z, &*s, &*zs];
-        let attachment_constraints = [
-            &[
-                constants::DEPTH_COMPONENT16,
-                constants::DEPTH_COMPONENT24,
-                constants::DEPTH_COMPONENT32F,
-                constants::DEPTH24_STENCIL8,
-                constants::DEPTH32F_STENCIL8,
-            ][..],
-            &[
-                constants::STENCIL_INDEX8,
-                constants::DEPTH24_STENCIL8,
-                constants::DEPTH32F_STENCIL8,
-            ][..],
-            &[constants::DEPTH_STENCIL][..],
-        ];
+        let attachment_constraints = match self.webgl_version {
+            WebGLVersion::WebGL1 => [
```

Comment:
If the Webgl2 constraints are a superset of the webgl1 constraints, we should be able to do:
```rust
let webgl1_attachment_constraints = &[[constants::DEPTH_COMPONENT16, ...][..], &[constants::STENCIL_INDEX8, ...][..], &[constants::DEPTH_STENCIL][..]];
let webgl2_attachment_constraints = &[[constants::DEPTH_STENCIL][..], &[constants::DEPTH_STENCIL][..], &[]];
let empty = &[];
let extra_constraints = match self.webgl_version {
    WebGLVersion::WebGL1 => empty.iter(),
    WebGLVersion::WebGL2 => webgl2_attachment_constraints.iter(),
};
let attachment_constraints = webgl1_attachment_constraints.iter().zip(extra_constraints).map(|(a, b)| a.iter().chain(b.iter()));
```
Then any code like check_attachment_constraints that iterates over the constraints can use `impl Iterator<&'a u32>`. I made a snippet at https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e0cda590f60315bf122372e123bdf318 to demonstrate.

Reply:
Thanks!